cpmpy/expressions/core.py

```diff
@@ -350,7 +351,7 @@ def __abs__(self):
         return Operator("abs", [self])
     # 'not' for now, no unary constraint for it
     def __invert__(self):
-        return (self == 0)
+        return Operator("not", [self])
```

Comment:
can do a typecheck on self.is_bool() here

also in Operator (or here?) you can overload it and check that it is not 'name == "not"' otherwise you can skip to remove the double negation and return self.args[0]